var/spack/repos/builtin/packages/openloops/package.py

```diff
@@ -0,0 +1,123 @@
+        "heftpphh", "heftpphhj", "heftpphhjj", "pphh2",
+        "pphhj2", "pphhjj2", "pphhjj_vbf", "bbhj", "heftpphj", "heftpphjj",
+        "heftpphjjj", "pphbb", "pphbbj", "pphj2",
+        "pphjj2", "pphjj_vbf", "pphjj_vbf_ew", "pphjjj2", "eetttt", "eettttj",
+        "eellllbb", "eett", "eett_ew", "eettj",
+        "eettjj", "eevtt", "eevttj", "eevttjj", "eevvtt", "eevvttj",
+        "eellll_ew", "eevv_ew", "eevvjj", "eell_ew", "eevjj",
+        "eehtt", "eehttj", "eehll_ew", "eehvtt", "eehhtt", "heftppllj",
+        "heftpplljj", "heftpplljjj")
+
+    variant('compile_extra', default=False,
+            description='Compile real radiation tree amplitudes')
+    variant('processes', description='Processes to install. See https://' +
+                                     'openloops.hepforge.org/process_' +
+                                     'library.php?repo=public for details',
+            values=disjoint_sets(('all.coll',), ('lhc.coll',), ('lcg.coll',),
+                                 all_processes).with_default('lhc.coll'))
+
+    variant('num_jobs', description='Number of parallel jobs to run. '  +
+                                    'Set to 1 if compiling a large number' +
+                                    'of processes (e.g. lcg.coll)', default=0)
+    depends_on('python', type=("build", "run"))
+
+    phases = ['configure', 'build', 'build_processes', 'install']
+
+    def configure(self, spec, prefix):
+        spack_env = ('PATH LD_LIBRARY_PATH CPATH C_INCLUDE_PATH' +
+                     'CPLUS_INCLUDE_PATH INTEL_LICENSE_FILE').split()
+        for k in env.keys():
+            if k.startswith('SPACK_'):
+                spack_env.append(k)
+
+        spack_env = ' '.join(spack_env)
+        is_intel = self.spec.satisfies('%intel')
+        njobs = self.spec.variants['num_jobs'].value
+
+        with open('openloops.cfg', 'w') as f:
+            print('[OpenLoops]', file=f)
+            print('import_env={0}'.format(spack_env), file=f)
+            print('num_jobs = {0}'.format(njobs), file=f)
+            print('cc = {0}'.format(env['SPACK_CC']), file=f)
+            print('cxx = {0}'.format(env['SPACK_CXX']), file=f)
+            print('fortran_compiler = {0}'.format(env['SPACK_FC']), file=f)
+            if self.spec.satisfies('@1.3.1') and not is_intel:
+                print('gfortran_f_flags = -ffree-line-length-none ', file=f)
+            if self.spec.satisfies('@2.1.1') and not is_intel:
+                print('gfortran_f_flags = -ffree-line-length-none ' +
+                      '-fdollar-ok -mcmodel=medium', file=f)
+
+        if self.spec.satisfies('@:1.999.999 processes=lcg.coll'):
+            copy(join_path(os.path.dirname(__file__), 'sft1.coll'), 'lcg.coll')
+        elif self.spec.satisfies('@2:2.999.999 processes=lcg.coll'):
+            copy(join_path(os.path.dirname(__file__), 'sft2.coll'), 'lcg.coll')
+
+    def build(self, spec, prefix):
+        scons = Executable('./scons')
```

Comment:
Interesting, scons is distributed with the source code?

Reply:
Yes